server/concept/ThingService.java

```diff
@@ -198,35 +192,97 @@ private void getPlayersByRoleType(Relation relation, UUID reqID) {
+        FunctionalIterator<? extends Thing> things;
+        if (getOwnersReq.hasThingType()) things = attribute.getOwners(getThingType(getOwnersReq.getThingType()));
+        else things = attribute.getOwners();
+        transactionSvc.stream(things, reqID, owners -> getOwnersResPart(reqID, owners));
     }
 
-    private void removePlayer(Relation relation, ConceptProto.Relation.RemovePlayer.Req removePlayerReq, UUID reqID) {
-        relation.removePlayer(getRoleType(removePlayerReq.getRoleType()), getThing(removePlayerReq.getPlayer()).asThing());
-        transactionSvc.respond(removePlayerRes(reqID));
+    private void requestNotSet(ConceptProto.Thing.Req thingReq, UUID uuid) {
+        throw TypeDBException.of(UNKNOWN_REQUEST_TYPE);
     }
 
-    private void getOwners(Attribute attribute, ConceptProto.Attribute.GetOwners.Req getOwnersReq, UUID reqID) {
-        FunctionalIterator<? extends Thing> things;
-        switch (getOwnersReq.getFilterCase()) {
-            case THING_TYPE:
-                things = attribute.getOwners(getThingType(getOwnersReq.getThingType()).asThingType());
-                break;
-            case FILTER_NOT_SET:
+    private static <T extends Concept> T notNull(@Nullable T concept) {
+        if (concept == null) throw TypeDBException.of(MISSING_CONCEPT);
+        return concept;
+    }
+
+    private ThingType getThingType(ConceptProto.ThingType protoThingType) {
+        switch (protoThingType.getTypeCase()) {
+            case ENTITY_TYPE: return getEntityType(protoThingType.getEntityType()).asThingType();
+            case RELATION_TYPE: return getRelationType(protoThingType.getRelationType()).asThingType();
+            case ATTRIBUTE_TYPE: return getAttributeType(protoThingType.getAttributeType()).asThingType();
+            case TYPE_NOT_SET:
             default:
-                things = attribute.getOwners();
+                throw TypeDBException.of(BAD_VALUE_TYPE, protoThingType.getTypeCase());
         }
+    }
 
-        transactionSvc.stream(things, reqID, owners -> getOwnersResPart(reqID, owners));
+    private EntityType getEntityType(ConceptProto.EntityType protoType) {
+        return conceptMgr.getEntityType(protoType.getLabel());
     }
 
+    private RelationType getRelationType(ConceptProto.RelationType protoType) {
+        return conceptMgr.getRelationType(protoType.getLabel());
+    }
+
+    private AttributeType getAttributeType(ConceptProto.AttributeType protoType) {
+        return conceptMgr.getAttributeType(protoType.getLabel());
+    }
+
+    private RoleType getRoleType(ConceptProto.RoleType protoRoleType) {
+        RelationType relationType = conceptMgr.getRelationType(protoRoleType.getScope());
+        if (relationType != null) return relationType.getRelates(protoRoleType.getLabel());
+        else return null;
+    }
+
+    private Thing getThing(ConceptProto.Thing.Req thingReq) {
+        return notNull(conceptMgr.getThing(ByteArray.of(thingReq.getIid().toByteArray())));
```

Comment:
I believe we had `getIid` as `getIID` capitalised before? That is more in line with our style